share/spice/tlds/TLDs.js

```diff
@@ -0,0 +1,43 @@
+(function(env) {
+    "use strict";
+    env.ddg_spice_tlds = function(api_result) {
+        if(!api_result) {
+            return Spice.failed('namecheap');
+        }
+        var item = api_result.ApiResponse.CommandResponse.gettldlist;
+        if(!item) {
+            return Spice.failed('namecheap');
+        }
+        /* extract data from JSON */
+        
+        var tldName = item.tldName;
+        var tldType = item.tldType;
+       
+        var data = {
+            tldType: tldType
+            tldName: tldName
+        };
+        Spice.add({
+            id: 'namecheap',
+            name: 'tldlist',
+            data: data,
```

Comment:
I'm a little confused -- does the API response contain an array or TLDs? If so I would suggest you simply pass along the array of results you need from the `api_response` to your `data`:

```js
var tld_list = api_result.ApiResponse.CommandRespoinse.gettldlist;
if (!tld_list){
    return Spice.failed('tlds');
}
Spice.add({
    id: 'tlds',
    name: 'TLDs' //this should really be an IA Page topic
    data: tld_list,
    normalize: function(item){
        title: item.tldName.
        subtitle: item.tldType
    },
    ...
}
...
```